examples/dag/worker.py

```diff
@@ -32,16 +33,16 @@ def step2(self, context: Context):
         }
 
     @hatchet.step(parents=["step1", "step2"])
-    def step3(self, context: Context):
-        one = context.step_output("step1")["rando"]
-        two = context.step_output("step2")["rando"]
+    def step3(self, context: Context) -> dict[str, int]:
+        one = cast(dict[str, Any], context.step_output("step1"))["rando"]
+        two = cast(dict[str, Any], context.step_output("step2"))["rando"]
```

Comment:
this is not ideal, but it's required afaict because of the two possible return types (either a pydantic model or something else). I think this kind of thing should be top of mind as we keep iterating on the API we expose